process/interceptors/multiDataInterceptor.go

```diff
@@ -113,32 +113,26 @@ func (mdi *MultiDataInterceptor) ProcessReceivedMessage(message p2p.MessageP2P,
 		return err
 	}
 
-	lastErrEncountered := error(nil)
-	wgProcess := &sync.WaitGroup{}
-	wgProcess.Add(len(multiDataBuff))
-
-	go func() {
-		wgProcess.Wait()
-		mdi.throttler.EndProcessing()
-	}()
-
+	listInterceptedData := make([]process.InterceptedData, len(multiDataBuff))
 	errOriginator := mdi.antifloodHandler.IsOriginatorEligibleForTopic(message.Peer(), mdi.topic)
-	allWhiteListed := true
-	for _, dataBuff := range multiDataBuff {
+
+	for index, dataBuff := range multiDataBuff {
 		var interceptedData process.InterceptedData
 		interceptedData, err = mdi.interceptedData(dataBuff, message.Peer(), fromConnectedPeer)
+		listInterceptedData[index] = interceptedData
 		if err != nil {
-			lastErrEncountered = err
-			wgProcess.Done()
-			continue
+			mdi.throttler.EndProcessing()
+			return err
 		}
 
 		isWhiteListed := mdi.whiteListRequest.IsWhiteListed(interceptedData)
 		if !isWhiteListed && errOriginator != nil {
-			lastErrEncountered = errOriginator
-			allWhiteListed = false
-			wgProcess.Done()
-			continue
+			mdi.throttler.EndProcessing()
+			log.Debug("got message from peer on topic only for validators", "originator",
```

Comment:
Might add a TODO to remove this in the future when we find out that this works correctly

Reply:
done